### PR TITLE
Add ability to log exceptions.

### PR DESCRIPTION
**Pull request by valentin-claras**
Add a new listener able to save exceptions through the request.
Add a new formatter able to log exceptions.

### CODE REVIEW

**Comment by fabdsp** [comment]
en attendant, :+1:

**Comment by gbouyge** [comment]
:+1:

**Comment by omansour** [comment]
:+1:  je viens de découvrir http://php.net/manual/fr/exception.gettraceasstring.php :crown:

File: README.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ Symfony Bundle to log Request/Response with Monolog.
 m6_web_log_bridge:
     resources:
         - %kernel.root_dir%/config/m6web_log_bridge.yml
-    content_formatter: m6web_log_bridge.log_content_provider # Provider service name
+    content_formatter: m6web_log_bridge.log_content_formatter # Provider service name
```
Comment (mikaelrandy) [inline README.md]: Is an upgrade without configuration update could broke an existing project ?

**Comment by fabdsp** [comment]
:+1:

File: README.md [hunk 2]
```diff
@@ -187,6 +187,32 @@ or simply override this parameter : ```m6web_log_bridge.log_content_formatter.cl
     Here response content
 
 
+## Logging exceptions
+
+The bundle allow detailed logging of exceptions. This is assured by the use of `exception.log` in the configuration.
```
Comment (mikaelrandy) [inline README.md]: s/This is assured/This is ensured/

File: README.md (hunk 1 above)
Comment (valentin-claras) [inline README.md]: I simply update the documentation since I could not find any reference to `log_content_provide`.

Comment (mikaelrandy) [inline README.md]: :+1:

File: src/M6Web/Bundle/LogBridgeBundle/Formatter/ExceptionFormatter.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     */
+    public function getLogContent(Request $request, Response $response, array $options)
+    {
+        if ($request->attributes->has($this->requestExceptionAttribute)) {
+            $exception = $request->attributes->get($this->requestExceptionAttribute);
+
+            return parent::getLogContent($request, $response, $options) . self::getExceptionTrace($exception);
+        }
+
+        return parent::getLogContent($request, $response, $options);
```
Comment (mikaelrandy) [inline src/M6Web/Bundle/LogBridgeBundle/Formatter/ExceptionFormatter.php]: I dont like the double return with same call in 2 cases.

I would like to see somthing like this : 
```php
$logContent = parent::getLogContent($request, $response, $options);

if ($request->attributes->has($this->requestExceptionAttribute)) {
    $exception = $request->attributes->get($this->requestExceptionAttribute);
 
    $logContent .= self::getExceptionTrace($exception);
}

return $logContent;
```

File: src/M6Web/Bundle/LogBridgeBundle/Formatter/ExceptionFormatter.php [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+     *
+     * @return string
+     */
+    protected function getExceptionTrace(\Exception $exception, $level = 1)
+    {
+        if (($previousException = $exception->getPrevious()) !== null) {
+            return self::formatException($exception, $level) . self::getExceptionTrace($previousException, $level + 1);
+        }
+
+        return self::formatException($exception, $level);
```
Comment (mikaelrandy) [inline src/M6Web/Bundle/LogBridgeBundle/Formatter/ExceptionFormatter.php]: same point.

**Comment by mikaelrandy** [comment]
:+1:

**Pull request merged**